### PR TITLE
Template set datatransport

### PR DESCRIPTION
**Pull request by moljac**
### Describe your contribution

1. Moved `Xamarin.Google.Android.DataTransport.*` to this repo from:
   https://github.com/xamarin/XamarinComponents/tree/main/Android/Google.Android.DataTransport 
2. created template set

### Does this change any of the generated binding API's?

`net6-android` TFM added

### Artifacts updated

Checklist
- [x] Add any metadata needed to get PR to compile
- [x] Verify any new namespaces are properly renamed (utilities.cake)
- [x] Verify nuget/namespace spelling (utilities.cake)

### CODE REVIEW

File: source/GooglePlayServicesProject.cshtml [hunk 1]
```diff
@@ -260,23 +266,32 @@
...
   <ItemGroup>
     <!-- PackageReference -->
-    @foreach (var dep in @Model.NuGetDependencies) {
-      if (!dep.IsProjectReference) {
-      <PackageReference Include="@(dep.NuGetPackageId)" Version="@(dep.NuGetVersion)" PrivateAssets="none" />
+    @foreach (var dep in @Model.NuGetDependencies) 
+    {
+      if (!dep.IsProjectReference) 
+      {
+        @if (@Model.NuGetPackageId == "Xamarin.Firebase.Perf" || @Model.NuGetPackageId == "Xamarin.Firebase.Datatransport")
```
Comment (jpobst) [inline source/GooglePlayServicesProject.cshtml:273]: What are these changes for?  Both of these packages list `androidx.annotation.annotation` in their POM so they should be picked up automatically.

File: utilities.cake [hunk 2]
```diff
@@ -173,6 +173,12 @@ Task ("spell-check")
                 "DiskLruCache",
                 "RecyclerViewIntegration",
                 "JavaX",
+                "DataTransport",
+                "TransportRuntime",
+                "DataTransport",
```
Comment (jpobst) [inline utilities.cake:178]: Duplicates.

File: source/GooglePlayServicesProject.cshtml (hunk 1 above)
Comment (moljac) [reply source/GooglePlayServicesProject.cshtml:273]: Nuget version package downgrade detection warnings/errors.

Let me see if I can remove that after cleanup.

Comment (moljac) [reply source/GooglePlayServicesProject.cshtml]: it works locally without that snippet. Let's wait CI

File: utilities.cake (hunk 2 above)
Comment (moljac) [reply utilities.cake]: removed

File: templates/datatransport/Project.cshtml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    {
+      if (dep.IsProjectReference) 
+      {
+    <ProjectReference Include="..\..\generated\@(dep.MavenArtifact.MavenGroupId).@(dep.MavenArtifact.MavenArtifactId)\@(dep.MavenArtifact.MavenGroupId).@(dep.MavenArtifact.MavenArtifactId).csproj" PrivateAssets="none" />
+      } 
+      else 
+      {
+        if (dep.NuGetPackageId == "Xamarin.AndroidX.Annotation")
+        {
+          <PackageReference Include="@(dep.NuGetPackageId)" Version="1.2.0.3" />
```
Comment (jpobst) [inline templates/datatransport/Project.cshtml:101]: What is this for?  Does the automatic dependency resolution have a bug?

Comment (moljac) [reply templates/datatransport/Project.cshtml:101]: transitive dependencies cause "Detected package downgrade" warnings and errors. Explicitly adding nuget with version is workaround for that

**Review by jpobst — COMMENTED** [review]
This removes a lot of API compared to the same version(s) on NuGet.  Do we know why?
```
# API diff: Xamarin.Google.Android.DataTransport.TransportRuntime.dll

## Xamarin.Google.Android.DataTransport.TransportRuntime.dll

### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.BuildConfig
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.EncodedPayload
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.EventInternal
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.IDestination
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.IEncodedDestination
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.TransportContext
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.TransportRuntime
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.TransportRuntimeComponent
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Backends


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Backends.BackendRegistryModule
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Backends.BackendRequest
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Backends.BackendResponse
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Backends.CreationContext
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Backends.IBackendFactory
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Backends.IBackendRegistry
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Backends.ITransportBackend
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Backends.TransportBackendDiscovery
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Dagger


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.BindsAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.BindsInstanceAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.BindsOptionalOfAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.ComponentAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IBinds
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IBindsInstance
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IBindsOptionalOf
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IComponent
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IComponentBuilder
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IComponentFactory
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.ILazy
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IMapKey
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IMembersInjector
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IModule
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IProvides
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.IReusable
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.ISubcomponent
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.ISubcomponentBuilder
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.ISubcomponentFactory
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MapKeyAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.ModuleAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.ProvidesAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.ReusableAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.SubcomponentAttribute
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.BetaAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.ComponentDefinitionTypeAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.DaggerCollections
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.GwtIncompatibleAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.IBeta
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.IComponentDefinitionType
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.IGwtIncompatible
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.IInjectedFieldSignature
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.InjectedFieldSignatureAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.MapBuilder
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.MembersInjectors
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.MemoizedSentinel
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.Preconditions
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.SetBuilder
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.ClassKeyAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.ElementsIntoSetAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.IClassKey
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.IElementsIntoSet
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.IIntKey
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.IIntoMap
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.IIntoSet
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.ILongKey
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.IMultibinds
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.IStringKey
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.IntKeyAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.IntoMapAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.IntoSetAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.LongKeyAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.MultibindsAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Dagger.MultiBindings.StringKeyAttribute
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Logging


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Logging.Logging
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Persistence


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Persistence.EventStoreModule
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Persistence.IEventStore
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Persistence.PersistedEvent
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Persistence.SQLiteEventStore
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Retries


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Retries.IFunction
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Retries.IRetryStrategy
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Retries.Retries
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Scheduling


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.DefaultScheduler
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.IScheduler
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.SchedulingConfigModule
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.SchedulingModule
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Scheduling.JobScheduling


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.JobScheduling.AlarmManagerScheduler
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.JobScheduling.AlarmManagerSchedulerBroadcastReceiver
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.JobScheduling.IWorkScheduler
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.JobScheduling.JobInfoScheduler
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.JobScheduling.JobInfoSchedulerService
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.JobScheduling.SchedulerConfig
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.JobScheduling.Uploader
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Scheduling.JobScheduling.WorkInitializer
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Synchronization


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Synchronization.ISynchronizationGuard
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Synchronization.ISynchronizationGuardCriticalSection
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Synchronization.SynchronizationException
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Time


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Time.IClock
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Time.IMonotonic
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Time.IWallTime
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Time.MonotonicAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Time.TestClock
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Time.TimeModule
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Time.UptimeClock
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Time.WallTimeAttribute
#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Time.WallTimeClock
### Removed Namespace Xamarin.Google.Android.DataTransport.Runtime.Util


#### Removed Type Xamarin.Google.Android.DataTransport.Runtime.Util.PriorityMapping
```

File: templates/datatransport/Project.cshtml (hunk 3 above)
Comment (moljac) [reply templates/datatransport/Project.cshtml:101]: didn't want to use something like:

```
 <PropertyGroup>
    <NoWarn>$(NoWarn);NU1605</NoWarn>
  </PropertyGroup>
```

**Comment by moljac** [comment]
> This removes a lot of API compared to the same version(s) on NuGet. Do we know why?

No.

I finished those bindings long time ago. I think 1st version almost 2 years (with/after AndroidX). The last few months ago. 
For template set I did not change metadata, just moved it over here. So, the only difference would be tools.

**Comment by moljac** [comment]
looks like jar or aar is missing.

**Comment by moljac** [comment]
seems that I deleted `LibraryProjectZip` when excluding `JavaSourcesJar` and `JavaDocJar`

**Comment by moljac** [comment]
> seems that I deleted `LibraryProjectZip` when excluding `JavaSourcesJar` and `JavaDocJar`

fixed

**Review by jpobst — APPROVED** [review]

**Pull request merged**